openupgrade_scripts/scripts/mail/16.0.1.10/pre-migration.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+_models_renames = [("mail.channel.partner", "mail.channel.member")]
+_tables_renames = [("mail_channel_partner", "mail_channel_member")]
+_columns_renames = {
+    "mail_mail": [("scheduled_date", None)],
+    "mail_message": [("add_sign", "email_add_signature")],
+}
+_columns_copies = {
+    "mail_channel_rtc_session": [
+        ("channel_partner_id", "channel_member_id", "integer")
+    ],
+}
+
+
+def mail_channel_member_ir_model_data(env):
+    """
+    Update ir.model.data name that changed. Otherwise Odoo tries to load
+    this data a second time, but it fails because of a unicity constraint
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE ir_model_data
+        SET name='channel_member_general_channel_for_admin'
+        WHERE name='channel_partner_general_channel_for_admin'
+        """,
+    )
+
+
+def ir_act_server_rename_state_email(env):
+    """
+    ir.actions.server state selection key 'email' is now 'mail_post'.
+    """
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE ir_act_server
+        SET state='mail_post'
+        WHERE state='email';
+        """,
+    )
+
+
+def ir_act_server_compute_mail_post_method(env):
+    """
+    Add mail_post_method column on ir_act_server and compute the values
+    """
```

Comment:
Not sure if a migration script is useful or if we let Odoo compute and store the values when uploading the module?
Same question for the other new field of the model, mail_post_autofollow

Reply:
hi @marielejeune. 
Thanks for your contribution on the OpenUpgrade project ! 
Regarding your question : If I understand correctly, your SQL request make the same things than this function : 

https://github.com/odoo/odoo/blob/16.0/addons/mail/models/ir_actions_server.py#L99

The unique reason to "precompute" a field in the pre-migration script, by SQL, is to speed up the computation. 

So, this happens when these two conditions are met:
1) ORM computation is very slower than the SQL request
2) the table has a big amount of rows. This second point of course depends on your instance, but generally, it occures on stock.move(.line), sale.order(.line), mail.message, account.move(.line), etc...

In your case : 
- computation done by ORM is trivial
- there is never "a lot of" rows in ir_act_server table.

So both pre-compute are unnecessary and can be removed.

Note : even if both conditions are met, you can always base your migration on ORM if it's OK for your migration (small database for exemple), and let another contributor improving by an SQL request if needed.
 See for exemple that PR https://github.com/OCA/OpenUpgrade/pull/3626 that improves 13-14 account migration that has been done in a second step.

Reply:
Clear, thanks! I will remove these two methods